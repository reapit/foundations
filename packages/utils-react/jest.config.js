const { pathsToModuleNameMapper } = require('ts-jest')
const { jestGlobalConfig } = require('@reapit/ts-scripts')
const { compilerOptions } = require('./tsconfig.json')

module.exports = {
  ...jestGlobalConfig,
  moduleNameMapper: {
    ...jestGlobalConfig.moduleNameMapper,
    ...pathsToModuleNameMapper(compilerOptions.paths, {
      prefix: '<rootDir>/',
    }),
  },
  coverageThreshold: {
    global: {
      branches: 59,
      functions: 83,
      lines: 81,
      statements: 82,
    },
  },
}
