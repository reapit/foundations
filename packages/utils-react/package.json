{
  "name": "@reapit/utils-react",
  "version": "0.0.1",
  "private": true,
  "description": "Common Package for Reapit Foundation",
  "repository": {
    "type": "git",
    "url": "git+git@github.com:reapit/foundations.git"
  },
  "license": "MIT",
  "author": "Reapit Ltd",
  "main": "./src/index.ts",
  "scripts": {
    "start": "echo '...skipping...'",
    "test": "cross-env TZ=UTC jest --watch --colors --silent",
    "build": "echo '...skipping...'",
    "lint": "eslint --cache --ext=ts,tsx,js src",
    "check": "tsc --noEmit --skipLibCheck",
    "release": "echo '...skipping...'",
    "deploy": "echo '...skipping...'",
    "publish": "echo '...skipping...'",
    "conf": "echo '...skipping...'",
    "commit": "yarn test --coverage --no-cache --silent --forceExit --detectOpenHandles --runInBand --watch=false && jest-coverage-badges --input src/tests/coverage/coverage-summary.json --output src/tests/badges && yarn lint --fix && yarn check"
  },
  "dependencies": {
    "@linaria/core": "^5.0.2",
    "@linaria/react": "^5.0.3",
    "@reapit/connect-session": "workspace:packages/connect-session",
    "@reapit/foundations-ts-definitions": "^1.5.9",
    "@reapit/utils-common": "workspace:packages/utils-common",
    "@sentry/node": "^7.107.0",
    "error-polyfill": "^0.1.3",
    "himalaya": "^1.1.0",
    "pell": "^1.0.6",
    "react-image-crop": "8.6.12",
    "uuid": "^8.3.2"
  },
  "devDependencies": {
    "@linaria/shaker": "^5.0.3",
    "@reapit/config-manager": "workspace:packages/config-manager",
    "@reapit/ts-scripts": "workspace:packages/ts-scripts",
    "@reapit/ts-types": "workspace:packages/ts-types",
    "@typescript-eslint/eslint-plugin": "^6.21.0",
    "@typescript-eslint/parser": "^6.21.0",
    "babel-jest": "^29.7.0",
    "concurrently": "^6.5.1",
    "cross-env": "^7.0.3",
    "eslint": "8.57.0",
    "eslint-plugin-prettier": "^5.1.3",
    "jest": "^29.7.0",
    "jest-coverage-badges": "^1.1.2",
    "serialize-error": "^7.0.1",
    "ts-jest": "^29.1.2",
    "typescript": "^5.4.2"
  }
}
