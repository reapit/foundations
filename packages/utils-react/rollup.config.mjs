import linaria from '@wyw-in-js/rollup'
import css from 'rollup-plugin-css-only'
import babel from '@rollup/plugin-babel'
import typescript from '@rollup/plugin-typescript'
import resolve from '@rollup/plugin-node-resolve'
import commonjs from '@rollup/plugin-commonjs'
import terser from '@rollup/plugin-terser'
import url from '@rollup/plugin-url'
import { appendFileSync } from 'fs'
import { execSync } from 'child_process'
import svgr from '@svgr/rollup'

export default [
  {
    input: './src/index.ts',
    output: {
      dir: './dist/cjs',
      format: 'cjs',
      sourcemap: true,
    },
    external: ['@linaria/core', '@linaria/react', 'react', 'react-dom', '@sentry/react'],
    plugins: [
      resolve({
        browser: true,
      }),
      commonjs(),
      typescript({
        tsconfig: './tsconfig.cjs.json',
      }),
      linaria({
        sourceMap: process.env.NODE_ENV !== 'production',
      }),
      css({
        output(styles) {
          execSync('mkdir ./dist')
          appendFileSync('./dist/index.css', styles)
        },
      }),
      babel({
        presets: [
          [
            '@babel/preset-env',
            {
              useBuiltIns: 'usage',
              corejs: 3,
            },
          ],
        ],
        extensions: ['.ts', '.tsx'],
        babelHelpers: 'runtime',
        plugins: ['@babel/plugin-transform-runtime'],
      }),
      url(),
      svgr({ icon: true }),
      terser(),
    ],
  },
  {
    input: './src/index.ts',
    output: {
      dir: './dist/esm',
      format: 'esm',
      sourcemap: true,
    },
    external: ['@linaria/core', '@linaria/react', 'react', 'react-dom', '@sentry/react'],
    plugins: [
      resolve({
        browser: true,
      }),
      commonjs(),
      typescript({
        tsconfig: './tsconfig.esm.json',
      }),
      linaria({
        sourceMap: process.env.NODE_ENV !== 'production',
      }),
      css({
        output: false,
      }),
      babel({
        presets: [
          [
            '@babel/preset-env',
            {
              useBuiltIns: 'usage',
              corejs: 3,
            },
          ],
        ],
        extensions: ['.ts', '.tsx'],
        babelHelpers: 'runtime',
        plugins: ['@babel/plugin-transform-runtime'],
      }),
      url(),
      svgr({ icon: true }),
      terser(),
    ],
  },
]
