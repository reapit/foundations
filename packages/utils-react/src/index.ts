export * from './validators'
export * from './rich-text-editor'
export * from './html-renderer'
export * from './image-cropper'
export * from './catch-chunk-error'
export * from './cloud-alert/ok-page'
export * from './demo/is-demo'
export * from './get-platform-headers'
export * from './logger'
export * from './switch-window-mode'
export * from './key-animation'
export * from './helper-widget'
export * from './use-async-state'
export * from './searchable-multi-select'
export * from './desktop-link'
export * from './login'
