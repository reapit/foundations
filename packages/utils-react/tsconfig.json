{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "typeRoots": ["./src/core/definitions.d.ts", "node_modules/@types", "../../node_modules/@types"],
    "baseUrl": "./",
    "sourceRoot": "/",
    "paths": {
      "@reapit/utils-common": ["../utils-common/src"],
      "@reapit/use-reapit-data": ["../use-reapit-data/src"]
    },
    "declaration": true
  },

  "include": [
    "src",
    "src/types/global.d.ts"
  ],
  "exclude": ["public", "dist", "src/scripts", "node_modules", "src/tests/coverage", "../../node_modules"]
}
