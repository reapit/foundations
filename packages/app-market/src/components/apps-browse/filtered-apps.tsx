import React, { FC } from 'react'
import { reapitConnectBrowserSession } from '../../core/connect-session'
import { GetActionNames, getActions } from '@reapit/utils-common'
import { useReapitGet } from '@reapit/utils-react'
import { Loader } from '@reapit/elements'
import { AppSummaryModelPagedResult } from '@reapit/foundations-ts-definitions'
import { useAppsBrowseState } from './use-apps-browse-state'

export const FilteredAppsCollection: FC = () => {
  const { appsBrowseFilterState } = useAppsBrowseState()
  const queryParams = appsBrowseFilterState ? appsBrowseFilterState : {}

  const [apps, appsLoading] = useReapitGet<AppSummaryModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions(window.reapit.config.appEnv)[GetActionNames.getApps],
    queryParams,
    fetchWhenTrue: [appsBrowseFilterState],
  })

  console.log('Filtered Apps: ', apps)

  if (appsLoading) return <Loader />

  return null
}
