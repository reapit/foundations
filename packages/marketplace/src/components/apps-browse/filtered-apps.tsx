import React, { FC, useMemo } from 'react'
import { reapitConnectBrowserSession } from '../../core/connect-session'
import { GetActionNames, getActions } from '@reapit/use-reapit-data'
import { useReapitGet } from '@reapit/use-reapit-data'
import {
  Card,
  Col,
  elFadeIn,
  FlexContainer,
  Grid,
  Loader,
  PersistentNotification,
  PlaceholderImage,
  elMl5,
} from '@reapit/elements'
import { AppSummaryModelPagedResult } from '@reapit/foundations-ts-definitions'
import { useReapitConnect } from '@reapit/connect-session'
import { cx } from '@linaria/core'
import { navigateRoute } from '../../utils/navigation'
import { RoutePaths } from '../../constants/routes'
import { useNavigate } from 'react-router-dom'
import { BrowseAppsSubtitle, cardCursor, IsFreeNotice } from './__styles__'
import { checkHasFilters } from './apps-browse'
import { useAppsBrowseState } from '../../core/use-apps-browse-state'
import { filterRestrictedAppsList } from '../../utils/browse-app'

export interface FiltersAppsCollectionProps {
  collectionId: string | null
}

export const FilteredAppsCollection: FC<FiltersAppsCollectionProps> = ({ collectionId }) => {
  const { appsBrowseFilterState, appsBrowseConfigState } = useAppsBrowseState()
  const navigate = useNavigate()
  const { connectSession } = useReapitConnect(reapitConnectBrowserSession)
  const clientId = connectSession?.loginIdentity.clientId
  const product = connectSession?.loginIdentity.orgProduct ?? 'agencyCloud'
  const hasFilters = useMemo(checkHasFilters(appsBrowseFilterState), [appsBrowseFilterState])
  const queryParams = hasFilters ? { ...appsBrowseFilterState, clientId, pageSize: 100, product } : {}
  const title = appsBrowseConfigState?.items.find((item) => item.id === collectionId)?.content?.title

  const [unfilteredApps, appsLoading] = useReapitGet<AppSummaryModelPagedResult>({
    reapitConnectBrowserSession,
    action: getActions[GetActionNames.getApps],
    queryParams,
    fetchWhenTrue: [hasFilters],
  })

  const apps = useMemo(filterRestrictedAppsList(unfilteredApps, connectSession), [unfilteredApps])

  if (appsLoading) return <Loader />

  return (
    <>
      {title && <BrowseAppsSubtitle>{title}</BrowseAppsSubtitle>}
      <Grid>
        {apps?.data?.map(({ id, name, summary, developer, isDirectApi, iconUri, isFree }) => (
          <Col key={id}>
            <Card
              className={cx(elFadeIn, cardCursor)}
              onClick={navigateRoute(navigate, `${RoutePaths.APPS_BROWSE}/${id}`)}
              hasMainCard
              mainCardHeading={name}
              mainCardSubHeading={
                <FlexContainer isFlexJustifyBetween isFlexAlignCenter>
                  {developer} {isFree && <IsFreeNotice className={elMl5}>FREE</IsFreeNotice>}
                </FlexContainer>
              }
              mainCardSubHeadingAdditional={isDirectApi ? 'Integration' : ''}
              mainCardBody={summary}
              mainCardAvatarUrl={iconUri ?? <PlaceholderImage placeholder="placeholderSmall" size={50} />}
            />
          </Col>
        ))}
      </Grid>
      {!appsLoading && apps?.data && Boolean(!apps.data.length) && (
        <PersistentNotification isInline isExpanded isFullWidth intent="primary">
          Unfortunately, there are &lsquo;No Results&rsquo; matching your search term or filtering options. Some of our
          categories have been recently added and our developers are updating their listings so, please come back and
          check again. Alternately, you could try searching by app or developer name.
        </PersistentNotification>
      )}
    </>
  )
}
