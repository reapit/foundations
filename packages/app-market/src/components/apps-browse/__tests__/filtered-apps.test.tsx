import { useReapitGet } from '@reapit/utils-react'
import React from 'react'
import { render } from '../../../tests/react-testing'
import { FilteredAppsCollection } from '../filtered-apps'

jest.mock('@reapit/utils-react', () => ({
  useReapitGet: jest.fn(() => [{}, false]),
  objectToQuery: jest.fn(() => '?query=string'),
}))

const mockUseReapitGet = useReapitGet as jest.Mock

describe('FilteredAppsCollection', () => {
  it('should match a snapshot', () => {
    expect(render(<FilteredAppsCollection />)).toMatchSnapshot()
  })

  it('should match a snapshot when loading', () => {
    mockUseReapitGet.mockReturnValue([null, true])
    expect(render(<FilteredAppsCollection />)).toMatchSnapshot()
  })
})
