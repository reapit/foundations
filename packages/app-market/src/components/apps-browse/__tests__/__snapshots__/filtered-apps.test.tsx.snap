// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilteredAppsCollection should match a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="el-snack-holder"
      />
      <div
        class="el-grid"
      >
        <div
          class="el-col"
        >
          <div
            class="el-fade-in card-cursor el-card-wrap"
          >
            <div
              class="el-card-main-wrap"
            >
              <div
                class="el-card-image-wrap"
              >
                <div
                  class="el-placeholder-image"
                  style="font-size: 2.1875em; width: 56px; height: 56px;"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 60 35"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M50.9685 13.7768C50.1805 8.8795 45.8082 5.12069 40.5469 5.12069C38.5676 5.12069 36.6656 5.65245 35.016 6.64427C32.5063 2.54863 27.9881 0 22.9688 0C15.2148 0 8.90625 6.12582 8.90625 13.6552C8.90625 13.7018 8.90625 13.7502 8.90801 13.7968C3.92461 14.6119 0 18.8308 0 23.8966C0 29.544 4.84816 34.1379 10.6641 34.1379H49.3359C55.1518 34.1379 60 29.544 60 23.8966C60 18.7876 56.0119 14.542 50.9685 13.7768Z"
                      fill="#E3E3E3"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="el-card-heading-wrap"
              >
                <h5
                  class="el-card-heading"
                >
                  MOCK_APP_NAME
                </h5>
                <h6
                  class="el-card-sub-heading"
                >
                  <div
                    class="el-flex-justify-between el-flex-align-center el-flex-container"
                  >
                     
                    <span
                      class="is-free-notice"
                    >
                      FREE
                    </span>
                  </div>
                </h6>
                <h6
                  class="el-card-sub-heading-additional"
                />
              </div>
              <button
                class="el-mobile-toggle"
              >
                <span
                  class="el-icon"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 12 12"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="el-card-body-wrap"
            >
              Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar
            </div>
          </div>
        </div>
        <div
          class="el-col"
        >
          <div
            class="el-fade-in card-cursor el-card-wrap"
          >
            <div
              class="el-card-main-wrap"
            >
              <div
                class="el-card-image-wrap"
              >
                <div
                  class="el-placeholder-image"
                  style="font-size: 2.1875em; width: 56px; height: 56px;"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 60 35"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M50.9685 13.7768C50.1805 8.8795 45.8082 5.12069 40.5469 5.12069C38.5676 5.12069 36.6656 5.65245 35.016 6.64427C32.5063 2.54863 27.9881 0 22.9688 0C15.2148 0 8.90625 6.12582 8.90625 13.6552C8.90625 13.7018 8.90625 13.7502 8.90801 13.7968C3.92461 14.6119 0 18.8308 0 23.8966C0 29.544 4.84816 34.1379 10.6641 34.1379H49.3359C55.1518 34.1379 60 29.544 60 23.8966C60 18.7876 56.0119 14.542 50.9685 13.7768Z"
                      fill="#E3E3E3"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="el-card-heading-wrap"
              >
                <h5
                  class="el-card-heading"
                >
                  MOCK_APP_NAME_ANOTHER
                </h5>
                <h6
                  class="el-card-sub-heading"
                >
                  <div
                    class="el-flex-justify-between el-flex-align-center el-flex-container"
                  >
                     
                    <span
                      class="is-free-notice"
                    >
                      FREE
                    </span>
                  </div>
                </h6>
                <h6
                  class="el-card-sub-heading-additional"
                />
              </div>
              <button
                class="el-mobile-toggle"
              >
                <span
                  class="el-icon"
                >
                  <svg
                    fill="none"
                    height="1em"
                    viewBox="0 0 12 12"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="el-card-body-wrap"
            >
              Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="el-snack-holder"
    />
    <div
      class="el-grid"
    >
      <div
        class="el-col"
      >
        <div
          class="el-fade-in card-cursor el-card-wrap"
        >
          <div
            class="el-card-main-wrap"
          >
            <div
              class="el-card-image-wrap"
            >
              <div
                class="el-placeholder-image"
                style="font-size: 2.1875em; width: 56px; height: 56px;"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 60 35"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M50.9685 13.7768C50.1805 8.8795 45.8082 5.12069 40.5469 5.12069C38.5676 5.12069 36.6656 5.65245 35.016 6.64427C32.5063 2.54863 27.9881 0 22.9688 0C15.2148 0 8.90625 6.12582 8.90625 13.6552C8.90625 13.7018 8.90625 13.7502 8.90801 13.7968C3.92461 14.6119 0 18.8308 0 23.8966C0 29.544 4.84816 34.1379 10.6641 34.1379H49.3359C55.1518 34.1379 60 29.544 60 23.8966C60 18.7876 56.0119 14.542 50.9685 13.7768Z"
                    fill="#E3E3E3"
                  />
                </svg>
              </div>
            </div>
            <div
              class="el-card-heading-wrap"
            >
              <h5
                class="el-card-heading"
              >
                MOCK_APP_NAME
              </h5>
              <h6
                class="el-card-sub-heading"
              >
                <div
                  class="el-flex-justify-between el-flex-align-center el-flex-container"
                >
                   
                  <span
                    class="is-free-notice"
                  >
                    FREE
                  </span>
                </div>
              </h6>
              <h6
                class="el-card-sub-heading-additional"
              />
            </div>
            <button
              class="el-mobile-toggle"
            >
              <span
                class="el-icon"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 12 12"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="el-card-body-wrap"
          >
            Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar
          </div>
        </div>
      </div>
      <div
        class="el-col"
      >
        <div
          class="el-fade-in card-cursor el-card-wrap"
        >
          <div
            class="el-card-main-wrap"
          >
            <div
              class="el-card-image-wrap"
            >
              <div
                class="el-placeholder-image"
                style="font-size: 2.1875em; width: 56px; height: 56px;"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 60 35"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M50.9685 13.7768C50.1805 8.8795 45.8082 5.12069 40.5469 5.12069C38.5676 5.12069 36.6656 5.65245 35.016 6.64427C32.5063 2.54863 27.9881 0 22.9688 0C15.2148 0 8.90625 6.12582 8.90625 13.6552C8.90625 13.7018 8.90625 13.7502 8.90801 13.7968C3.92461 14.6119 0 18.8308 0 23.8966C0 29.544 4.84816 34.1379 10.6641 34.1379H49.3359C55.1518 34.1379 60 29.544 60 23.8966C60 18.7876 56.0119 14.542 50.9685 13.7768Z"
                    fill="#E3E3E3"
                  />
                </svg>
              </div>
            </div>
            <div
              class="el-card-heading-wrap"
            >
              <h5
                class="el-card-heading"
              >
                MOCK_APP_NAME_ANOTHER
              </h5>
              <h6
                class="el-card-sub-heading"
              >
                <div
                  class="el-flex-justify-between el-flex-align-center el-flex-container"
                >
                   
                  <span
                    class="is-free-notice"
                  >
                    FREE
                  </span>
                </div>
              </h6>
              <h6
                class="el-card-sub-heading-additional"
              />
            </div>
            <button
              class="el-mobile-toggle"
            >
              <span
                class="el-icon"
              >
                <svg
                  fill="none"
                  height="1em"
                  viewBox="0 0 12 12"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M11.0498 3.17368L6.00273 8.55662L0.956931 3.17508C0.738006 2.94162 0.383056 2.94165 0.164132 3.1751C-0.054715 3.40862 -0.0547118 3.78713 0.164149 4.02062L5.60635 9.82494C5.82527 10.0584 6.1802 10.0583 6.39913 9.82493L11.8413 4.02064L11.8426 4.01921C12.0577 3.78166 12.0516 3.40311 11.8288 3.17369C11.6116 2.94991 11.2671 2.94989 11.0498 3.17368Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="el-card-body-wrap"
          >
            Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar Foo bar
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FilteredAppsCollection should match a snapshot when loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="el-snack-holder"
      />
      <div
        class="el-loader-container"
      >
        <div
          class="el-loader"
        >
          <div
            class="el-loader-moving-bar"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="el-snack-holder"
    />
    <div
      class="el-loader-container"
    >
      <div
        class="el-loader"
      >
        <div
          class="el-loader-moving-bar"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
